releases/releases.go

```diff
@@ -0,0 +1,146 @@
+package releases
+
+import (
+  "fmt"
+  "os"
+  "path"
+  "regexp"
+  "strings"
+
+  "github.com/speakeasy-api/sdk-generation-action/internal/environment"
+)
+
+type ReleasesInfo struct {
+  ReleaseVersion         string
+  OpenAPIDocVersion      string
+  SpeakeasyVersion       string
+  OpenAPIDocPath         string
+  PythonPackagePublished bool
+  PythonPackageName      string
+  PythonPackageURL       string
+  PythonPath             string
+  NPMPackagePublished    bool
+  NPMPackageName         string
+  NPMPackageUrl          string
+  TypescriptPath         string
+  GoPackagePublished     bool
+  GoPackageURL           string
+  GoPath                 string
+}
+
+func (r ReleasesInfo) String() string {
+  releasesOutput := []string{}
+
+  if r.NPMPackagePublished {
+    releasesOutput = append(releasesOutput, fmt.Sprintf("- [NPM v%s] https://www.npmjs.com/package/%s/v/%s - %s", r.ReleaseVersion, r.NPMPackageName, r.ReleaseVersion, r.TypescriptPath))
+  }
+
+  if r.PythonPackagePublished {
+    releasesOutput = append(releasesOutput, fmt.Sprintf("- [PyPI v%s] https://pypi.org/project/%s/%s - %s", r.ReleaseVersion, r.PythonPackageName, r.ReleaseVersion, r.PythonPath))
+  }
+
+  if r.GoPackagePublished {
+    repoPath := os.Getenv("GITHUB_REPOSITORY")
+    releasesOutput = append(releasesOutput, fmt.Sprintf("- [Go v%s] https://github.com/%s/releases/tag/v%s - %s", r.ReleaseVersion, repoPath, r.ReleaseVersion, r.GoPath))
+  }
+
+  if len(releasesOutput) > 0 {
+    releasesOutput = append([]string{"\n### Releases"}, releasesOutput...)
+  }
+
+  return fmt.Sprintf(`%s## Version %s
+### Changes
+Based on:
+- OpenAPI Doc %s %s
+- Speakeasy CLI %s https://github.com/speakeasy-api/speakeasy%s`, "\n\n", r.ReleaseVersion, r.OpenAPIDocVersion, r.OpenAPIDocPath, r.SpeakeasyVersion, strings.Join(releasesOutput, "\n"))
+}
+
+func UpdateReleasesFile(releaseInfo ReleasesInfo) error {
+  releasesPath := getReleasesPath()
+
+  f, err := os.OpenFile(releasesPath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0o600)
+  if err != nil {
+    return fmt.Errorf("error opening releases file: %w", err)
+  }
+  defer f.Close()
+
+  _, err = f.WriteString(releaseInfo.String())
+  if err != nil {
+    return fmt.Errorf("error writing to releases file: %w", err)
+  }
+
+  return nil
+}
+
+var (
+  releaseInfoRegex = regexp.MustCompile(`(?s)## Version (\d+\.\d+\.\d+)\n### Changes\nBased on:\n- OpenAPI Doc (\d+\.\d+\.*\d*) (.*?)\n- Speakeasy CLI (\d+\.\d+\.\d+).*?`)
+  npmReleaseRegex  = regexp.MustCompile(`- \[NPM v\d+\.\d+\.\d+\] (https:\/\/www.npmjs.com\/package\/(.*?)\/v\/\d+\.\d+\.\d+) - (.*)`)
+  pypiReleaseRegex = regexp.MustCompile(`- \[PyPI v\d+\.\d+\.\d+\] (https:\/\/pypi.org\/project\/(.*?)\/\d+\.\d+\.\d+) - (.*)`)
+  goReleaseRegex   = regexp.MustCompile(`- \[Go v\d+\.\d+\.\d+\] (.*?) - (.*)`)
```

Comment:
The only real change to this file besides moving it was that I added an additional regex group to each of these release regexes in order to retrieve the package url. It is used to populate the `GoPackageURL`, `NPMPackageURL`, etc. fields i added to `ReleasesInfo`